common/src/main/scala/react/floatingui/syntax.scala

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package react.floatingui
+
+import japgolly.scalajs.react.*
+import japgolly.scalajs.react.vdom.html_<^.*
+import lucuma.ui.syntax.all.given
+
+import scala.scalajs.js
+
+object syntax:
+  // extension (tag: VdomTag)
+  //   def withTooltip(tooltip: VdomNode, placement: Placement = Placement.Top): VdomNode =
+  //     Tooltip(trigger = tag, tooltip = tooltip, placement = placement)
+  //   def withOptionalTooltip(
+  //     tooltip:   js.UndefOr[VdomNode],
+  //     placement: Placement = Placement.Top
+  //   ): TagMod =
+  //     tooltip.fold(tag)(tt => Tooltip(trigger = tag, tooltip = tt, placement = placement))
+
+  extension (node: VdomNode)
+    def withTooltip(tooltip: VdomNode, placement: Placement = Placement.Top): VdomNode =
+      Tooltip(trigger = <.span(node), tooltip = tooltip, placement = placement)
+
+    def withOptionalTooltip(
```

Comment:
`withOptionalTooltip` could be renamed `withTooltip` and would work for both cases. I started out with `Option[VdomNode` for the `tooltip` parameter, but the `js.UndefOr` allows either a `VdomNode` or a `Option[VdomNode]` to be used. More importantly, it allows the use of `Option[String]`. 

I kind of like that the name indicates in the code that the tooltip may not be present, but it's probably best to just have the one method. Thoughts?